exchanges/gemini/gemini_websocket.go

```diff
@@ -52,179 +58,266 @@ func (g *Gemini) WsSubscribe(dialer *websocket.Dialer) error {
 	for i, c := range enabledCurrencies {
 		val := url.Values{}
 		val.Set("heartbeat", "true")
-
-		endpoint := fmt.Sprintf(g.Websocket.GetWebsocketURL(),
+		endpoint := fmt.Sprintf("%s%s/%s?%s",
+			g.WebsocketURL,
+			geminiWsMarketData,
 			c.String(),
 			val.Encode())
-
-		conn, _, err := dialer.Dial(endpoint, http.Header{})
+		conn, conStatus, err := dialer.Dial(endpoint, http.Header{})
 		if err != nil {
-			return err
+			return fmt.Errorf("%v %v %v Error: %v", endpoint, conStatus, conStatus.StatusCode, err)
 		}
-
-		go g.WsReadData(conn, c, geminiWsMarketData)
-
+		go g.WsReadData(conn, c)
 		if len(enabledCurrencies)-1 == i {
 			return nil
 		}
-
-		time.Sleep(5 * time.Second) // rate limiter, limit of 12 requests per
+		time.Sleep(time.Minute) // rate limiter, limit of 1 requests per
```

Comment:
Its really rate limited that beast mode?

Reply:
`For public WebSocket APIs, we recommend that you do not exceed 1 request per symbol per minute.`
I'll have another check to see if this sleep actually does anything to help with that